### PR TITLE
Update Affiliated Editors' role descriptions to include their future tasks under the pyOpenSci partnership

### PR DESCRIPTION
**Pull request by dhomeier**
Part of the transition of our Affiliated Package Review process into [partnership with pyOpenSci](https://pyopensci.discourse.group/t/astropy-partnership-next-steps/432/6)

Fixing #569

### CODE REVIEW

File: roles.json [hunk 1]
```diff
@@ -231,12 +231,11 @@
...
         "responsibilities": {
-            "description": "Oversee the affiliated package submission and review process, including:",
+            "description": "Oversee affiliated packages in partnership with <a href='https://www.pyopensci.org/software-peer-review/partners/scientific-communities.html'>pyOpenSci</a>, including:",
             "details": [
+                "Representing Astropy as pyOpenSci Software Review Editor",
                 "Keeping track of submissions to the affiliated package ecosystem",
-                "Managing the review process for packages",
-                "Making decisions regarding accepting and rejecting packages based on reviews",
-                "Organizing the re-review of packages that have not been recently reviewed"
+                "Guiding submissions tthrough the pre-submission and submission steps to the <a href='https://www.pyopensci.org/software-peer-review/how-to/author-guide.html#submit-your-package-for-peer-review'>Peer Review system</a>"
```
Comment (pllim) [inline roles.json:238]:
```suggestion
                "Guiding submissions through the pre-submission and submission steps to the <a href='https://www.pyopensci.org/software-peer-review/how-to/author-guide.html#submit-your-package-for-peer-review'>Peer Review system</a>"
```

**Review by pllim — COMMENTED** [review]
I think the content is good, except for a typo.

Would be nice if @hamogu and @lwasser could have a look through. Thanks!

**Review by hamogu — APPROVED** [review]

**Review by pllim — APPROVED** [review]
This is much better than the version I came up with over at #573 .

When should we merge this?

**Review by ceb8 — APPROVED** [review]

**Review by pllim — APPROVED** [review]
We already started the transition here and there, and it has been 2 weeks with multiple approvals and no objection, so I am merging. Thanks!

**Pull request merged**